autoreduce_qp/queue_processor/reduction/process_manager.py

```diff
@@ -27,33 +25,89 @@ def __init__(self, message: Message, run_name: str) -> None:
     def run(self) -> Message:
         """Run the reduction subprocess."""
         try:
+            # Create a temp directory to store reduction result
+            # Volume bind for the container to write to and host to read from
+            temp_dir = tempfile.TemporaryDirectory()  # pylint: disable=consider-using-with
+            os.chmod(temp_dir.name, 0o777)
+
             # We need to run the reduction in a new process, otherwise scripts
             # will fail when they use things that require access to a main loop
             # e.g. a GUI main loop, for matplotlib or Mantid
-            python_path = sys.executable
-            with tempfile.NamedTemporaryFile("w+") as temp_output_file:
-                args = [python_path, RUNNER_PATH, self.message.serialize(), temp_output_file.name, self.run_name]
-                logger.info("Calling: %s %s %s %s %s", python_path, RUNNER_PATH,
-                            self.message.serialize(limit_reduction_script=True), temp_output_file.name, self.run_name)
+            serialized_vars = self.message.serialize()
+            serialized_vars_truncated = self.message.serialize(limit_reduction_script=True)
+            args = ["python3", "runner.py", serialized_vars, self.run_name]
+            logger.info("Calling: %s %s %s %s ", "python3", "runner.py", serialized_vars_truncated, self.run_name)
+
+            # Return a client configured from environment variables
+            # The environment variables used are the same as those used by the Docker command-line client
+            # https://docs.docker.com/engine/reference/commandline/cli/#environment-variables
+            client = docker.from_env()
+
+            # Create a container and run it. Equivalent to docker run.
+            # To build runner-mantid image, in same directory as Dockerfile run:
+            # docker build -t runner-mantid .
 
-                # Copy and update the subprocess environment to inherit the
-                # parent one, and append the PYTHONPATH of the queue_processor module
-                environment = os.environ.copy()
+            # Pull image, containers.create doesn't pull everytime
+            client.images.pull('ghcr.io/autoreduction/runner-mantid:6.2.0')
+            mount = f'{os.path.expanduser("~")}/.autoreduce/dev/test-archive'
+            if not os.path.exists(mount):
+                Path(mount).mkdir(parents=True, exist_ok=True)
+            os.chmod(mount, 0o777)
 
-                pythonpath = environment.get("PYTHONPATH", "")
-                pythonpath += f':{RUNNER_PATH.split("autoreduce_qp")[0]}'
-                pythonpath = environment["PYTHONPATH"] = pythonpath
+            Path(f'{os.path.expanduser("~")}/.autoreduce/dev/reduced-data').mkdir(parents=True, exist_ok=True)
```

Comment:
This should switch between `~/.autoreduce/dev/reduced-data` and `/instrument` depending on `os.environ["AUTOREDUCTION_PRODUCTION"]`, though it might be better to extract the base of that path in https://github.com/autoreduction/autoreduce-utils/blob/main/autoreduce_utils/settings.py#L46